### PR TITLE
feat(api): add milestones endpoint to explorer API

### LINKED ISSUE #576: Add a `milestones/` endpoint with filters to the Explorer API
It would be very nice if we could return a list of milestones that has the following properties

* filterable by `startTimestamp` and `endTimestamp`
* sortable by `newest` and `oldest`
* pagination with `pageSize`

### PR DESCRIPTION
**Pull request by Alexandcoats**
Closes #576

### CODE REVIEW

File: docs/api-explorer.yml [hunk 1]
```diff
@@ -74,6 +74,35 @@ paths:
...
+      responses:
+        "200":
+          description: Successful operation.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/BalanceResponse"
+              examples:
+                default:
+                  $ref: "#/components/examples/balance-example"
```
Comment (Alex6323) [inline docs/api-explorer.yml:99]: Update summary and description and response refs with `MilestonesResponse`

File: docs/api-explorer.yml [hunk 2]
```diff
@@ -297,6 +326,20 @@ components:
...
+        type: integer
+      example: 1662139730
+      description: Start timestamp for filtering.
+    endTimestamp:
+      in: query
+      name: endTimestamp
+      schema:
+        type: integer
+      example: 1662139830
+      description: Start timestamp for filtering.
```
Comment (Alex6323) [inline docs/api-explorer.yml:342]: ~~Start~~ End

File: src/bin/inx-chronicle/api/stardust/explorer/routes.rs [hunk 3]
```diff
@@ -164,3 +166,38 @@ async fn block_children(
...
+    // If any record is left, use it to make the paging state
+    let cursor = record_stream.try_next().await?.map(|rec| {
+        MilestonesCursor {
+            milestone_index: rec.index,
+            page_size,
+        }
+        .to_string()
+    });
+
+    Ok(MilestonesResponse { items, cursor })
```
Comment (Alex6323) [inline src/bin/inx-chronicle/api/stardust/explorer/routes.rs:202]: That might be wrong! What happens with the successfully collected `items` if `try_next` on the `record_stream` fails (i.e. no record is left) ?

File: docs/api-explorer.yml (hunk 1 above)
Comment (Alexandcoats) [reply docs/api-explorer.yml:99]: Wow, clearly I was very tired when I made these changes.

**Pull request merged**